convey/conveymetric/metric.go

```diff
@@ -0,0 +1,76 @@
+package conveymetric
+
+import (
+	"fmt"
+	"github.com/Comcast/webpa-common/convey"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/provider"
+	"regexp"
+	"strings"
+)
+
+const UnknownLabel = "unknown"
+
+type MetricClosure func()
+
+type CMetric interface {
+	Update(data convey.C) (MetricClosure, error)
+}
+
+var (
+	validKeyRegex   = regexp.MustCompile(`^[a-zA-Z_:]+$`)
+	replaceRegex    = regexp.MustCompile(`[\W]+`)
+	underscoreRegex = regexp.MustCompile(`_+`)
+)
+
+func NewConveyMetric(provider provider.Provider, tag string, name string) CMetric {
+	return &cMetric{
+		provider: provider,
+		tag:      tag,
+		name:     name,
+		metrics:  make(map[string]metrics.Gauge),
+	}
+}
+
+type cMetric struct {
+	provider provider.Provider
+	tag      string
+	name     string
+	metrics  map[string]metrics.Gauge
```

Comment:
Don't use a map here.  Use the label API.  Have a single `metric metrics.Gauge` struct member.  Then, use a label for each model:

```
cMetric.metric.With("model", value).Add(1.0)
```

The gauge will then have to be configured with the "model" label.